site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -342,6 +342,168 @@ If you add it to your own file, you should write something like
+   });
+   
+It's actually not that simple as it will emit JavaScript objects
+but any functions they reference must be expressible by calling `toString`.
+   
+The implications of that are as follows
+
+-  You can not use ES6 object shorthand function syntax.
+-  You can not reference/bind other functions directly.
+-  You can not use a closure directly.
+-  You can not use ES6 arrow syntax.
+-  You can not use other function transformations like currying.
+
+.. code-block:: javascript
+
+   mergeInto(LibraryManager.library, {
+     // bad: ES6 object shorthand function syntax not allowwed
+     bad_01() {
+       alert('hi');
+     },
+     
+     // bad: Can not bind/reference other functions directly.
+     bad_02: document.querySelector.bind(document),
+     
+     // bad: You can not use a closure directly
+     bad_03: (function() {
+       var callCount = 0;
+       return function() {
+         console.log("times called: ", ++callCount);
+       };
+     }()),
+     
+     // bad: You can not use ES6 arrow syntax
+     bad_04: () => { console.log("hi!"); },
+     
+     // bad: You can not curry/transform
+     bad_05: curry(scrollTo, 0),  
+   });
+   
+Workarounds to the above issues
+
+You can put code in another file using a ``<script>`` tag 
+OR ``--pre-js`` OR ``--post-js``. Of course many of these
+features work deeper in a function but the function directly 
+attached to the object passed to ``mergeInto`` must be a plain old 
+vanilla JavaScript ES5 function.
+
+One way to workaround some of these issues is to have a function
+replace itself and have it called during initialization.
+
+.. code-block:: javascript
+
+   mergeInto(LibraryManager.library, {
+
+     // Can not bind/reference other functions directly workround
+     good_02__postset: '_good_02();',
+     good_02: function() {
+       _good_o2 = document.querySelector.bind(document);
+     },
+     
+     // Can not use a closure workaround
+     good_03__postset: '_good_03();',
+     good_03: function() {
+       var callCount = 0;
+       _good_03 = function() {
+         console.log("times called: ", ++callCount);
+       };
+     },
+     
+     // Can not curry/transform workaround
+     good_05__postset: '_good_05();',
+     good_05: function() {
+       _good_05 = curry(scrollTo, 0);  
+    },
+
+   });
+
+This effectively emits this code
+
+.. code-block:: javascript
+
+     function _good_02() {
+       _good_o2 = document.querySelector.bind(document);
+     }
+     
+     function _good_03() {
+       var callCount = 0;
+       _good_03 = function() {
+         console.log("times called: ", ++callCount);
+       };
+     }
+     
+     function _good_05() {
+       _good_05 = curry(scrollTo, 0);  
+    };
+    
+    // Call each function once so it will replace itself
+    _good_02();
+    _good_03();
+    _good_05();
+
+You can also put most of your code in the ``xxx__postset`` strings. 
```

Comment:
It works in node 4.1 LTS (the current stable version of node) and since emscripten is just outputting the string what it outputs doesn't use the multi-line feature.

That brings up a good point through. If the only thing you're getting by evaluating the JavaScript at compile time is being able to only include library functions that are used then it would be nice to be able to just do

```
mergeInto(LibraryManager.library, {
   my_js__literal: `some code`,
});
```

and be able to avoid the dummy function and post set stuff. You get the same benefit of being able to trim the library at compile time without the extra code and self setting etc ....